### ISSUE TITLE
Updated analysis: Add additional subtype figures to immune deconv

### ISSUE THREAD

**Issue by jharenza** [AlexsLemonade/OpenPBTA-analysis#965]
#### What analysis module should be updated and why?
immune deconv

The overall figures are nice, but by lumping histologies, we are not able to detect subtype differences within histology. We know this because of a recent analysis on medulloblastoma subtypes enabling us to see large differences between subtypes. I suggest we keep the figures we have, but add one heatmap figure per broad histology with rows by molecular subtype so that we can try to detect differences within histologies.

#### What changes need to be made? Please provide enough detail for another participant to make the update.
Add one heatmap figure per broad histology with rows being molecular subtype.

#### What input data should be used? Which data were used in the version being updated?
update to use v18 data

#### When do you expect the revised analysis will be completed?
one week

#### Who will complete the updated analysis?
@komalsrathi

**Comment by komalsrathi** [comment]
@jharenza Can you confirm that there are only 7 broad histologies for which molecular_subtype info is available? Also can you check the attached plot? 

[heatmap_xCell_by_molecular_subtype.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/6224639/heatmap_xCell_by_molecular_subtype.pdf)

**Comment by komalsrathi** [comment]
@jharenza I was initially using the `molecular_subtype` field from the `data/pbta-histologies.tsv` file, which worked fine as it  has a 1-1 mapping for `Kids_First_Biospecimen_ID` and `molecular_subtype`. When I instead switched to using the file under `molecular-subtyping-pathology/results/compiled_molecular_subtypes.tsv`, it seems to have duplicates (not sure if this is a known issue and how to tackle it):

```
   Kids_First_Biospecimen_ID        molecular_subtype
1                BS_5P1TN10Z       CNS Embryonal, NOS
2                BS_5P1TN10Z   HGG, H3 G35, TP53 loss
3                BS_BFDEZK1C       CNS Embryonal, NOS
4                BS_BFDEZK1C              DMG, H3 K28
5                BS_C6ZZ3FY5       CNS Embryonal, NOS
6                BS_C6ZZ3FY5   HGG, H3 G35, TP53 loss
7                BS_G16NM5WJ              DMG, H3 K28
8                BS_G16NM5WJ    EPN, To be classified
9                BS_J8VX4D17              DMG, H3 K28
10               BS_J8VX4D17    EPN, To be classified
11               BS_JXF8A2A6                      EWS
12               BS_JXF8A2A6 CRANIO, To be classified
13               BS_KSKZ9J7J            CNS HGNET-MN1
14               BS_KSKZ9J7J    EPN, To be classified
15               BS_MB7WN0ZB       CNS Embryonal, NOS
16               BS_MB7WN0ZB              DMG, H3 K28
17               BS_N6N147BY      ETMR, C19MC-altered
18               BS_N6N147BY    HGG, To be classified
19               BS_P8WN8XEQ       CNS Embryonal, NOS
20               BS_P8WN8XEQ   HGG, H3 G35, TP53 loss
21               BS_STNH7YSX       CNS Embryonal, NOS
22               BS_STNH7YSX   HGG, H3 G35, TP53 loss
23               BS_TV5B86ZD       CNS Embryonal, NOS
24               BS_TV5B86ZD   HGG, H3 G35, TP53 loss
```

**Comment by jharenza** [comment]
Thanks, @komalsrathi - actually, the file we want to use is [this one](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/molecular-subtyping-pathology/results/compiled_molecular_subtypes_with_clinical_pathology_feedback.tsv), which corrects these manually after review.

**Comment by jharenza** [comment]
>I think it might be better to hold off on this until the compiled_molecular_subtypes_with_clinical_pathology_feedback.tsv has all RNA-Seq BS ids.

We can update this after v19 release #867 , at which point you can use the updated `pbta-histologies.tsv` file.

**Comment by jharenza** [comment]
@komalsrathi this is ready to be worked on now that #1026 was merged

**Comment by jharenza** [comment]
Closed with #1055
